x/mongo/driver/operation/hello_test.go

```diff
@@ -0,0 +1,838 @@
+package operation
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/assert"
+	"go.mongodb.org/mongo-driver/version"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+const documentSize = 5         // 5 bytes to start and end a document
+const embeddedDocumentSize = 7 // 7 bytes to append a document element
+const stringElementSize = 7    // 7 bytes to append a string element
+const int32ElementSize = 6     // 6 bytes to append an int32 element
+
+func assertAppendClientMaxLen(t *testing.T, got bsoncore.Document, wantD bson.D, maxLen int) {
+	t.Helper()
+
+	tooLarge := len(got)-documentSize > maxLen
+	assert.False(t, tooLarge, "got document is too large: %v", got)
+
+	wantBytes, err := bson.Marshal(wantD)
+	assert.Nil(t, err, "error marshaling want document: %v", err)
+
+	want := bsoncore.Document(wantBytes)
+
+	wantElems, err := want.Elements()
+	assert.Nil(t, err, "error getting elements from want document: %v", err)
+
+	// Compare element by element.
+	gotElems, err := got.Elements()
+	assert.Nil(t, err, "error getting elements from got document: %v", err)
+
+	areEqual := reflect.DeepEqual(gotElems, wantElems)
+	assert.True(t, areEqual, "got %v, want %v", gotElems, wantElems)
+}
+
+func encodeWithCallback(t *testing.T, cb func(int, []byte) ([]byte, error)) bsoncore.Document {
+	t.Helper()
+
+	var err error
+	idx, dst := bsoncore.AppendDocumentStart(nil)
+
+	dst, err = cb(len(dst), dst)
+	assert.Nil(t, err, "error appending client metadata: %v", err)
+
+	dst, err = bsoncore.AppendDocumentEnd(dst, idx)
+	assert.Nil(t, err, "error appending document end: %v", err)
+
+	got, _, ok := bsoncore.ReadDocument(dst)
+	assert.True(t, ok, "error reading document")
+
+	return got
+}
+
+func addMaxLenStringElem(key, name string) func() int {
+	return func() int {
+		return stringElementSize + len(key) + len(name)
+	}
+}
+
+//nolint:unparam
+func addMaxLenInt32Elem(key string) func() int {
```

Comment:
The "unparam" linter seems to complain that the `addMaxLenInt32Elem` function always receives the same input:
> x/mongo/driver/operation/hello_test.go:65:25: `addMaxLenInt32Elem` - `key` always receives `"memory_mb"` (unparam)

Optional: Consider changing one of the inputs to `addMaxLenInt32Elem` to satisfy the linter rather than disabling it.

Reply:
I don't think you can change any of the inputs, they are all "memory_mb." If that value is changed, test will fail expecting `{ "memory_mb": "x" }`.

Reply:
Sounds reasonable to keep it.